pkg/pool-aclamm/contracts/AclAmmPool.sol

```diff
@@ -180,6 +189,47 @@ contract AclAmmPool is BalancerPoolToken, PoolInfo, Version, IBasePool, BaseHook
         return _lastTimestamp;
     }
 
+    function setNewQ0(uint256 newQ0, uint256 startTime, uint256 endTime) external {
```

Comment:
it should receive the new sqrtQ0, so you don't need to calculate the sqrt later. Especially because this is an 18 decimal number, Math.sqrt won't work.